### DISCUSSION TITLE
HA Support

### DISCUSSION THREAD

**Discussion by flaming-archer** [ExpediaGroup/waggle-dance#285] [General]
**Is your feature request related to a problem? Please describe.**
Is there a solution that can be supported, such as switching to another service for execution if one service hangs up.

**Describe alternatives you've considered**
Or do  load balancing mysql

**Comment by patduin** [comment]
We run this in either AWS ECS or EKS, so Kubernetes with multiple pods and a load balancer in front.
Have a look at these projects Waggle Dance is a part of them they focus on the deployment you might find some pointers in there:
https://github.com/ExpediaGroup/apiary-federation
https://github.com/ExpediaGroup/apiary-data-lake


Load balancing MySQL: Waggle Dance doesn't use any RDS, so I think you mean the connection from Hive to an MySQL DB that's outside of the scope of this project.

**Comment by flaming-archer** [comment]
> We run this in either AWS ECS or EKS, so Kubernetes with multiple pods and a load balancer in front. Have a look at these projects Waggle Dance is a part of them they focus on the deployment you might find some pointers in there: https://github.com/ExpediaGroup/apiary-federation https://github.com/ExpediaGroup/apiary-data-lake
> 
> Load balancing MySQL: Waggle Dance doesn't use any RDS, so I think you mean the connection from Hive to an MySQL DB that's outside of the scope of this project.

Sorry for replay later. 
There are many HMSs in our scenario, and we plan to deploy many waggle instances as well. If one of the waggle instances hangs, the client can automatically switch to another waggle instance. How can we achieve this? Our environment has nothing to do with the cloud, it's all physical machines.

**Comment by githubjianli** [comment]
@flaming-archer Are you trying to play multiple HMS service in a single connection? like:
`thrift://wd-service-1:48869;wd-service-2:48869`

Or you are looking for WD accept multiple HMS endpoint when you have HA enabled  ?

**Reply by flaming-archer** [reply]
yes，like this `thrift://wd-service-1:48869;wd-service-2:48869`, Does it surpport in hive ,presto spark, flink or like this ?
